add sha to report and code to check it

mmulholla opened the pull request:
Calculates a sha value for a report and adds it to the report in the tool metadata.

```
apiversion: v1
kind: verify-report
metadata:
    tool:
        verifier-version: 1.9.0
        report-digest: uint64:3746509530969125134
```  

when the chart-verifier report option is used to generate a summary of a given report the ```report-digest``` value is compared to an equivalent digest value calculated for the report to make sure they match. If they do not match an error is output:
```Digest in report does not match report content, report has been modified.```

To support this a change was also made to include a GetVersion api and the release file was moved:

from: ```cmd/release/release_info.json```
to:  ```pkg/chartverifier/version/version_info.json```

documentation and scripts were modified to reflect this change.

To prevent failures with older reports the digest check is only performed if the report version is 1.9.0 or greater (current version is 1.8.0 and this digest feature is expected to GA in 1.9.0) 

Note: Because of the 1.9.0 check this change should not cause regressions in existing tests, until after 1.9.0 is released.

isutton commented on docs/helm-chart-checks.md:
````diff
@@ -32,6 +32,7 @@ Helm chart checks are a set of checks against which the Red Hat Helm chart-verif
   - Alternatively the ```--write-to-file``` flag can be used to write to a ```report.yaml``` file. 
 - An error log is created for all verify commands but can be optionally suppressed.
 - You can indicate that a chart is not to be published in the OpenShift catalog.
+- From chart verifier version 1.9.0 the generated report includes a sha value based on the report content. This is used during the submission process to verify the egrity of the report.
````
```suggestion
- From chart verifier version 1.9.0 the generated report includes a sha value based on the report content. This is used during the submission process to verify the integrity of the report.
```

mmulholla replied:
Updated. thx

isutton commented on pkg/chartverifier/report/types.go:
```diff
@@ -26,6 +26,7 @@ type ReportMetadata struct {
 type ToolMetadata struct {
 	Version                    string  `json:"verifier-version" yaml:"verifier-version"`
 	Profile                    Profile `json:"profile" yaml:"profile"`
+	ReportDigest               string  `json:"report-digest" yaml:"report-digest"`
```
Even though other keys are using `kebab-case` names, `snakeCase` should be used instead.

mmulholla replied:
Updated.

tisutisu commented on pkg/chartverifier/reportsummary/test-reports/report.json:
```diff
@@ -1 +1 @@
-{"apiversion":"v1","kind":"verify-report","metadata":{"tool":{"verifier-version":"1.7.0","profile":{"vendorType":"partner","version":"v1.1"},"chart-uri":"https://github.com/redhat-certification/chart-verifier/blob/main/tests/charts/psql-service/0.1.9/psql-service-0.1.9.tgz?raw=true","digests":{"chart":"sha256:9f4d290c9a89f2900c3ffe9c1680c2ac2e61a34619e526de4652e2f2dbf4609b","package":"89589ab22f1 ...
+{"apiversion":"v1","kind":"verify-report","metadata":{"tool":{"verifier-version":"1.9.0","profile":{"vendorType":"partner","version":"v1.1"},"report-digest":"uint64:6132714415651448453","chart-uri":"https://github.com/redhat-certification/chart-verifier/blob/main/tests/charts/psql-service/0.1.9/psql-service-0.1.9.tgz?raw=true","digests":{"chart":"sha256:9f4d290c9a89f2900c3ffe9c1680c2ac2e61a34619e ...
```
Same for this file as well.

mmulholla replied:
Done

tisutisu commented on docs/helm-chart-checks.md:
````diff
@@ -32,6 +32,7 @@ Helm chart checks are a set of checks against which the Red Hat Helm chart-verif
   - Alternatively the ```--write-to-file``` flag can be used to write to a ```report.yaml``` file. 
 - An error log is created for all verify commands but can be optionally suppressed.
 - You can indicate that a chart is not to be published in the OpenShift catalog.
+- From chart verifier version 1.9.0 the generated report includes a sha value based on the report content. This is used during the submission process to verify the egrity of the report.
````
`egrity` a type here.

mmulholla replied:
Thx.

mmulholla commented on pkg/chartverifier/reportsummary/test-reports/predigest/report.json:
```diff
@@ -0,0 +1 @@
+{"apiversion":"v1","kind":"verify-report","metadata":{"tool":{"verifier-version":"1.8.0","profile":{"vendorType":"partner","version":"v1.1"},"chart-uri":"https://github.com/redhat-certification/chart-verifier/blob/main/tests/charts/psql-service/0.1.9/psql-service-0.1.9.tgz?raw=true","digests":{"chart":"sha256:9f4d290c9a89f2900c3ffe9c1680c2ac2e61a34619e526de4652e2f2dbf4609b","package":"89589ab22f1 ...
```
Done.

mmulholla commented on pkg/chartverifier/report/report.go:
```diff
@@ -142,3 +147,42 @@ func loadReportFromRemote(url *url.URL) (string, error) {
+func (r *Report) GetReportDigest() (string, error) {
+
+	savedDigest := r.Metadata.ToolMetadata.ReportDigest
+	r.Metadata.ToolMetadata.ReportDigest = ""
```
the digest is based on a report with the digest set to blank, so need to blank it put before calculating its digest. Without this adding the digest would change the digest.

mmulholla commented:
Have added updates for code review comments but also made additional changes.

- Added an option to the report command and api to skip the digest check when generating a report summary but making it clear that is is intended for testing purposes only (David's request). Obviously the chart workflow will not set the option to skip the option will be useful in some of our tests. 
- Moved the code which performs the digest check to the report summary code to support the new option.

tisutisu reviewed (commented):
LGTM

tisutisu reviewed (approved):
All the review comments are addressed now.
